Source/WebCore/platform/network/ResourceResponseBase.cpp

```diff
@@ -81,11 +81,34 @@ ResourceResponseBase::ResourceResponseBase(const URL& url, const String& mimeTyp
     , m_isRedirected(false)
     , m_isRangeRequested(false)
     , m_isNull(false)
-    , m_usedLegacyTLS(UsedLegacyTLS::No)
-    , m_wasPrivateRelayed(WasPrivateRelayed::No)
-    , m_tainting(Tainting::Basic)
-    , m_source(Source::Unknown)
-    , m_type(Type::Default)
+{
+}
+
+ResourceResponseBase::ResourceResponseBase(std::optional<ResourceResponseBase::ResponseData> data)
+    : m_url(data.has_value() ? data->m_url : URL { })
+    , m_mimeType(data.has_value() ? data->m_mimeType : AtomString { })
+    , m_expectedContentLength(data.has_value() ? data->m_expectedContentLength : 0)
+    , m_textEncodingName(data.has_value() ? data->m_textEncodingName : AtomString { })
+    , m_httpStatusText(data.has_value() ? data->m_httpStatusText : AtomString { })
+    , m_httpVersion(data.has_value() ? data->m_httpVersion : AtomString { })
+    , m_httpHeaderFields(data.has_value() ? data->m_httpHeaderFields : HTTPHeaderMap { })
+    , m_networkLoadMetrics(data.has_value() ? data->m_networkLoadMetrics : Box<WebCore::NetworkLoadMetrics> { })
+    , m_certificateInfo(data.has_value() ? data->m_certificateInfo : std::nullopt)
+    , m_haveParsedCacheControlHeader(false)
```

Comment:
Can these be a an initializer in the header instead?  They doesn't depend on data.